File: crescer/java/trabalhos/tcc/api/src/main/java/br/com/cwi/crescer/api/service/GetUserService.java
package br.com.cwi.crescer.api.service;

import br.com.cwi.crescer.api.controller.response.PostResponse;
import br.com.cwi.crescer.api.controller.response.UserAccountResponse;
import br.com.cwi.crescer.api.mapper.UserResponseMapper;
import br.com.cwi.crescer.api.model.UserAccount;
import br.com.cwi.crescer.api.service.security.FindUserAuthenticatedService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GetUserService {
    @Autowired
    private FindUserAuthenticatedService findUserAuthenticatedService;
    @Autowired
    private UserResponseMapper userResponseMapper;

    public  UserAccountResponse get() {
        UserAccount user = findUserAuthenticatedService.getUser();
        return  userResponseMapper.toResponse(user);
    }
}

File: crescer/trabalhos/healthCare.api/src/main/kotlin/com/healthCare/api/service/user/GetUserService.kt
package com.healthCare.api.service.user

import com.healthCare.api.controller.response.UserAccountResponse
import com.healthCare.api.mapper.UserMapper
import org.springframework.stereotype.Service


@Service
class GetUserService(
    private val findUserAuthenticatedService: FindUserAuthenticatedService,
    private val userResponseMapper: UserMapper
) {

    fun get(): UserAccountResponse {
        val user = findUserAuthenticatedService.getUser()
        return userResponseMapper.toResponse(user)
    }
}
